### PR TITLE
PATCH /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by itskatherine**
-Added PATCH /api/reviews/:review_id endpoint
-Added 400 and 404 error handling for invalid ids and votes

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -72,4 +72,59 @@ describe("/api/reviews/:review_id", () => {
...
+    const req = { inc_votes: "katherine" };
+    return request(app)
+      .patch("/api/reviews/3")
+      .send(req)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid data type.");
+      });
+  });
+});
```
Comment (bainesface) [inline __tests__/app.test.js:153]: One more test here please 😁 

What if the user tries to send a request body that doesn't contain the key `inc_votes`?

File: models/reviews.model.js [hunk 2]
```diff
@@ -18,4 +18,43 @@ const fetchReviewById = (id) => {
...
+      WHERE review_id = ${id}
+      RETURNING *;
+      `;
+      return db.query(updateVoteQueryStr).then((updatedReviewObj) => {
+        const updatedReview = updatedReviewObj.rows[0];
+
+        return updatedReview;
+      });
+    });
+};
```
Comment (bainesface) [inline models/reviews.model.js:49]: Try to refactor here. I'm not sure what the first request to the database is doing that couldn't be handled by the second update request. 

Even though you are in an update request, do keep the prevention of sql injection in mind as your second query is vulnerable

File: __tests__/app.test.js (hunk 1 above)
Comment (itskatherine) [reply __tests__/app.test.js:153]: I've added this now, I assume it would return and unchanged review

File: models/reviews.model.js (hunk 2 above)
Comment (itskatherine) [reply models/reviews.model.js:49]: Great I think I've covered this points now!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -127,4 +127,27 @@ describe("PATCH /api/reviews/:review_id", () => {
         expect(response.body.msg).toBe("Invalid data type.");
       });
   });
+
+  test("400: Provided with a request that does not contain the inc_votes returns unchanged review object", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: change to 200

**Review by popatre — APPROVED** [review]

**Pull request merged**